interface-definitions/protocols-pim.xml.in

```diff
@@ -41,8 +57,134 @@
                   </constraint>
                 </properties>
               </leafNode>
+              <node name="igmp">
+                <properties>
+                  <help>Internet Group Management Protocol (IGMP) options</help>
+                </properties>
+                <children>
+                  <leafNode name="version">
+                    <properties>
+                      <help>Interface IGMP version</help>
+                      <valueHelp>
+                        <format>2</format>
+                        <description>IGMP version 2</description>
+                      </valueHelp>
+                      <valueHelp>
+                        <format>3</format>
+                        <description>IGMP version 3</description>
+                      </valueHelp>
+                      <constraint>
+                        <validator name="numeric" argument="--range 2-3"/>
+                      </constraint>
+                    </properties>
+                    <defaultValue>3</defaultValue>
+                  </leafNode>
+                </children>
+              </node>
+              <leafNode name="passive">
+                <properties>
+                  <help>Disable sending and receiving PIM control packets on the interface</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+              #include <include/source-address-ipv4.xml.i>
             </children>
           </tagNode>
+          <node name="ecmp">
+            <properties>
+              <help>Enable PIM ECMP</help>
+            </properties>
+            <children>
+              <leafNode name="rebalance">
+                <properties>
+                  <help>Enable PIM ECMP Rebalance</help>
+                  <valueless/>
+                </properties>
+              </leafNode>
+            </children>
+           </node>
+           <node name="igmp">
+            <properties>
+              <help>Internet Group Management Protocol (IGMP) options</help>
+            </properties>
+            <children>
+              <leafNode name="watermark-warn">
```

Comment:
What about the full name `watermark-warning`?

Reply:
Good idea, just copied that out of FRR. Will change in the next revision